api/pkg/service/auth/auth_http_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/ikawaha/goahttpcheck"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/h2non/gock.v1"
+
+	auth "github.com/tektoncd/hub/api/gen/auth"
+	server "github.com/tektoncd/hub/api/gen/http/auth/server"
+	"github.com/tektoncd/hub/api/pkg/testutils"
+)
+
+func LoginChecker(tc *testutils.TestConfig) *goahttpcheck.APIChecker {
+	checker := goahttpcheck.New()
+	checker.Mount(
+		server.NewLoginHandler,
+		server.MountLoginHandler,
+		auth.NewLoginEndpoint(New(tc)))
+	return checker
+}
+
+func TestLogin_Http(t *testing.T) {
+	tc := testutils.Setup(t)
+	testutils.LoadFixtures(t, tc.FixturePath())
+
+	defer gock.Disable()
```

Comment:
Can we create a function for gock as it used in both the tests i.e. valid and invalid code ?

Reply:
but the code is different for both the test..

Reply:
response for mock is different

Reply:
Okay got it :+1: